### ISSUE TITLE
Bug: Background flow loss assertion in m2o_fluctuating_lossless test uses incorrect expected value

### ISSUE THREAD

**Issue by ediwibowo-msft** [sonic-net/sonic-mgmt#24992]
### Is it platform specific

generic

### Importance or Severity

Low

### Description of the bug

In `tests/snappi_tests/pfc/files/m2o_fluctuating_lossless_helper.py`, the `verify_m2o_fluctuating_lossless_result` function asserts that each Background Flow has an average of ~10% loss:

```python
pytest_assert((abs(background_loss/4) - 10) < 1,
              "Each Background Flow must have an avg of 10% loss ")
```

Three issues with this assertion:

1. **Operator-precedence bug (silently ineffective).** `abs()` is applied only to `background_loss/4`, not to the deviation from 10. The check is effectively `(avg_loss - 10) < 1`, which silently passes for any value below 11 — including 0% loss. A totally broken test (e.g., no background drops at all) would still pass.

2. **Hardcoded `/4` divisor.** The divisor does not adapt if `background_flow_count` changes.

3. **Expected value is assuming the same scheduler weithts.** The test configures 4 Background Flows at 20% line rate each plus Test Flows totaling 30% line rate. The oversubscription is:
   - Total BG traffic at egress: `4 × 20% = 80%` of line rate
   - Total lossless traffic (PFC-protected, cannot be dropped): `20% + 10% = 30%`
   - Total demand: `110%` → only `100%` can exit
   - Excess to be dropped: `10%` of line rate, entirely from lossy BG traffic

   A naive "perfectly uniform drop" model gives `10%`, but that ignores how the egress scheduler actually allocates bandwidth. The real per-BG-flow loss is determined by the **per-queue DWRR scheduler weights** read from CONFIG_DB (`QUEUE` + `SCHEDULER`). Two examples from real DUTs:

   | DUT Sample| Lossless Qs (Q3/Q4) | Lossy Qs (Q0/Q1/Q2/Q5/Q6) | Analytical per-BG loss | IxNetwork-measured |
   | ------------------------- | ------------------- | ------------------------- | ---------------------- | ------------------ |
   | DUT-1   | weight 15           | weight 14                 | ~11.27%                | 11.259%            |
   | DUT-2          | weight 15           | weight 15                 | 10.0%                  | —                  |

   So `10%` is the wrong as static constant — the expected value must be derived from the actual scheduler weights on the egress port.

### Error Messages

No runtime error — the bug silently makes the assertion ineffective. A regression that produces 0% background loss would still pass this check.

### Impact

- The assertion does not actually validate the intended behavior.
- Regressions in PFC behavior on the lossy class could go undetected by this test.
- On platforms whose DWRR weights produce a true loss outside the `< 11` window (e.g., > 11.27% on different weights), a corrected absolute-deviation check against a static constant would fail spuriously.

### Steps to Reproduce

```bash
cd <sonic-mgmt repo>/tests && \
python3 -m pytest snappi_tests/pfc/test_m2o_fluctuating_lossless_with_snappi.py \
  --inventory=../ansible/veos,../ansible/ixia --host-pattern=all \
  --testbed=<testbed_name> \
  --testbed_file=../ansible/testbed.yaml \
  --topology tgen,any --skip_sanity
```

### Actual Behavior

The assertion `(abs(background_loss/4) - 10) < 1` passes for any average background loss `< 11%`, including 0%. The `/4` is hardcoded and the target (10%) does not match the analytical value, which is weight-dependent.

### Expected Behavior

The assertion should:

1. Compute `avg_loss = background_loss / background_flow_count` based on the actual number of background flows.
2. Derive the **expected** per-Background-Flow loss at runtime from the egress port's DWRR scheduler weights (`QUEUE` + `SCHEDULER` in CONFIG_DB) and the test/background flow rates — not from a hard-coded constant.
3. Compare the absolute deviation against a small tolerance: `abs(avg_loss - expected_bg_loss_percent) < TOLERANCE`.

### Relevant log output

N/A (silent assertion bug).

### Output of `show version`

N/A — bug is in the test framework, not platform-specific.

### Attach files (if any)

Flow statistics of sample DUT-1 with different weights (lossless 15 and lossy 14):
<img width="920" height="281" alt="image" src="https://github.com/user-attachments/assets/93c04607-c786-4c7a-9904-a1d46765ab79" />
